File: python_C_mysql/booklist.py
import pymysql


def main():
    host = "localhost"
    port = 3306
    database = "madangdb"
    username = "root"
    password = "0000"

    config = True
    try:
        conn = pymysql.connect(host=host, port=port, db=database, user=username
        , password=password, charset="utf8")
        print("연결 성공")
    except: # ignore error
        print("연결 실패")
        config = False

    if config:
        cursor = conn.cursor()
        cursor.execute("SELECT * FROM Book")
        books = cursor.fetchall()
        for book in books:
            print(book)
        cursor.close()
        conn.close()
if __name__ == "__main__":
    main()   
                  
                



    

File: python_C_mysql/booklist.c
#include <stdio.h>
#include <mysql.h>
#include <string.h>
// sudo apt install libmysql++
int main() {
    MYSQL *conn;
    MYSQL_RES  *res;
    MYSQL_ROW row;
    char *host = "localhost";
    char *user = "root";
    char *passwd = "0000";
    char *db = "madangdb";
    char query[255];
    int pot = 3306;
    
    strcpy (query, "select * from Book");

    conn = mysql_init(NULL);
    if(mysql_real_connect(conn, host, user, passwd, db, pot, NULL, 0)) {
        printf("MySQL 연결 성공\n");
    } else {
        printf("MySQL 연결 실패\n");
    }
    if(mysql_query(conn, query)) {
        printf("쿼리 실패\n");
    }
    

    res = mysql_store_result(conn);
    while(row = mysql_fetch_row(res)) {
        printf("%s\t", row[0]);
        printf("%s\t", row[1]);
        printf("%s\t", row[2]);
        printf("%s\t\n", row[3]);
    }
    mysql_close(conn);
    return 0;
}